src/UNIV2LPOracle.t.sol

```diff
@@ -546,6 +549,33 @@ contract UNIV2LPOracleTest is DSTest {
         assertTrue(daiEthLPOracle.pass());                           // Fail pass
     }
 
+    // This test examines the gas costs of a common pattern which needs
+    // to be maximally gas-optimized.
+    function test_gas_pass_poke() public {
+        uint256 preGas;
+        uint256 diffGas;
+
+        require(daiEthLPOracle.pass());
+
+        preGas = gasleft();
+        if (daiEthLPOracle.pass()) {
+            daiEthLPOracle.poke();
+        }
+        diffGas = preGas - gasleft();
+        assertTrue(diffGas <= 88408);  // Will need to be updated for the Berlin hardfork.
+        log_named_uint("pass+poke gas", diffGas);
```

Comment:
![Screenshot from 2021-03-23 09-22-32](https://user-images.githubusercontent.com/16065447/112161695-61be3c80-8bb9-11eb-9109-d0888a51bde4.png)

Reply:
Are you running with `make test` (or './test.sh`)? And are you on the latest `dapptools` (0.32.1)? Brian made me update the vars used to pass compile information (see [discussion](https://github.com/makerdao/univ2-lp-oracle/pull/41#discussion_r598849272) in this PR), so if you're on a version that doesn't recognize the new ones, it won't work for you.

Reply:
> Are you running with `make test` (or `./test.sh`)?

Yes

> And are you on the latest `dapptools` (0.32.1)?

The latest-latest is 0.32.2. I'm on that one.

If there are additional env vars to be set, shouldn't they be in `./test.sh`?

Btw, I just ran it again and got slightly better results:

```
e18r@jj:~/code/univ2-lp-oracle$ git log --oneline | head -n 1
65bf919 simplify pass_poke test
e18r@jj:~/code/univ2-lp-oracle$ git diff
e18r@jj:~/code/univ2-lp-oracle$ dapp --version
dapp 0.32.2
solc, the solidity compiler commandline interface
Version: 0.6.7+commit.b8d736ae.Linux.g++
hevm 0.45.0
e18r@jj:~/code/univ2-lp-oracle$ make test MATCH=test_gas_pass_poke
./test.sh test_gas_pass_poke
make[1]: Entering directory '/home/e18r/code/univ2-lp-oracle'
DAPP_BUILD_OPTIMIZE=1 DAPP_BUILD_OPTIMIZE_RUNS=0 dapp --use solc:0.6.12 build
+ dapp clean
+ rm -rf out
[...]
contract UNIV2LPOracleTest is DSTest {
^ (Relevant source part starts here and spans across multiple lines).

make[1]: Leaving directory '/home/e18r/code/univ2-lp-oracle'
Running 1 tests for src/UNIV2LPOracle.t.sol:UNIV2LPOracleTest
[FAIL] test_gas_pass_poke()
[...]
  
  «Assertion failed»
  pass+poke gas: 88676
  «Assertion failed»
  pass-only gas: 3776
```

Reply:
If you're on `0.32.2`, maybe `DAPP_BUILD_OPTMIZE_RUNS` actually works in that one. Set it to 200 in the `build` target of the Makefile and you should get the values in the tests.

I define "latest" by whatever the install script installs, which when I ran it a few days ago was `0.32.1` (although maybe it has changed).

There's still a debate here btw about where the optimizer runs value should be set (see my conversation w/Brian in this PR). Separate from that is the question handling dapptools versioning.

Reply:
Yes, in this version it actually works!

And btw, with 200 runs the babylonian sqrt is better, but with 2000 it is not.

Reply:
Good news, looks like DappHub has updated the version installed by the script...unfortunately, I'm still going to have to invent a way to check the `dapptools` version here, otherwise someone might try to deploy with a version that doesn't understand one or both of the new envvars.

Btw, on the sqrt, thing, are you sure you didn't just hit the known flaky case (input of zero)? It's fixed on `master`.